tests/sentry/rules/processing/test_delayed_processing.py

```diff
@@ -64,9 +100,50 @@ def test_bulk_fetch_events(self):
         pass
 
 
-class GetConditionGroupResultsTest(TestCase):
+class GetConditionGroupResultsTest(CreateEventTestCase):
```

Comment:
It looks like we're only testing the happy path here, let's make unit tests to validate each branch in logic. this is also magnified by the number of parameters as each param could use a test as well.

### Example
Function to test:
https://github.com/getsentry/sentry/blob/1f7741e0847c031157946907e7862ecdf77943ee/src/sentry/rules/processing/delayed_processing.py#L90-L101


tests: https://github.com/getsentry/sentry/blob/1f7741e0847c031157946907e7862ecdf77943ee/tests/sentry/rules/processing/test_delayed_processing.py#L111-L132

In those tests, i'm testing a number of different loop combinations, and results. each time i use a different context there's a `__with_` that then explains the thing i'm isolating in those tests.

by doing this, we can simplify the tests themselves (moving a lot of scaffolding into a `setUp` function). It also allows us to quickly identify issues in the future, if a test name includes the function and context for the logic it's testing, it's much faster to find the issue.

---

The end goal of these tests are to ensure that our code works in a variety of scenarios;

So for this code:
https://github.com/getsentry/sentry/blob/1f7741e0847c031157946907e7862ecdf77943ee/src/sentry/rules/processing/delayed_processing.py#L168-L207

we should have a _lot_ of tests...
- ensure each of the params works in edge cases
 - condition_groups is empty list
 - project is correct for group, project is incorrect for group

Then start to look at the branches in the code itself; where are there `if` statements, where are there loops, etc. For each of those pieces of logic, we should have a test to verify it.

Reply:
parameterizing does not work with unittest sad 😔